interfaces/builtin/steam_support.go

```diff
@@ -0,0 +1,214 @@
+mount options=(rw, rbind) /oldroot/etc/host.conf -> /newroot/etc/host.conf,
+mount options=(rw, rbind) /oldroot/etc/hosts -> /newroot/etc/hosts,
+mount options=(rw, rbind) /oldroot/**/*resolv.conf -> /newroot/etc/resolv.conf,
+mount options=(rw, rbind) /bindfile* -> /newroot/etc/timezone,
+
+mount options=(rw, rbind) /oldroot/run/systemd/journal/socket -> /newroot/run/systemd/journal/socket,
+mount options=(rw, rbind) /oldroot/run/systemd/journal/stdout -> /newroot/run/systemd/journal/stdout,
+
+mount options=(rw, rbind) /oldroot/usr/share/fonts/ -> /newroot/run/host/fonts/,
+mount options=(rw, rbind) /oldroot/usr/local/share/fonts/ -> /newroot/run/host/local-fonts/,
+mount options=(rw, rbind) /oldroot/{var/cache/fontconfig,usr/lib/fontconfig/cache}/ -> /newroot/run/host/fonts-cache/,
+mount options=(rw, rbind) /oldroot/home/**/.cache/fontconfig/ -> /newroot/run/host/user-fonts-cache/,
+mount options=(rw, rbind) /bindfile* -> /newroot/run/host/font-dirs.xml,
+
+mount options=(rw, rbind) /oldroot/usr/share/icons/ -> /newroot/run/host/share/icons/,
+mount options=(rw, rbind) /oldroot/home/**/.local/share/icons/ -> /newroot/run/host/user-share/icons/,
+
+mount options=(rw, rbind) /oldroot/run/user/[0-9]*/wayland-* -> /newroot/run/pressure-vessel/wayland-*,
+mount options=(rw, rbind) /oldroot/tmp/.X11-unix/X* -> /newroot/tmp/.X11-unix/X99,
+mount options=(rw, rbind) /bindfile* -> /newroot/run/pressure-vessel/Xauthority,
+
+mount options=(rw, rbind) /bindfile* -> /newroot/run/pressure-vessel/pulse/config,
+mount options=(rw, rbind) /oldroot/run/user/[0-9]*/pulse/native -> /newroot/run/pressure-vessel/pulse/native,
+mount options=(rw, rbind) /oldroot/dev/snd/ -> /newroot/dev/snd/,
+mount options=(rw, rbind) /bindfile* -> /newroot/etc/asound.conf,
+mount options=(rw, rbind) /oldroot/run/user/[0-9]*/bus -> /newroot/run/pressure-vessel/bus,
+
+mount options=(rw, rbind) /oldroot/run/dbus/system_bus_socket -> /newroot/run/dbus/system_bus_socket,
+mount options=(rw, rbind) /oldroot/run/systemd/resolve/io.systemd.Resolve -> /newroot/run/systemd/resolve/io.systemd.Resolve,
+mount options=(rw, rbind) /bindfile* -> /newroot/run/host/container-manager,
+
+# Allow masking of certain directories in the sandbox
+mount fstype=tmpfs options=(rw, nosuid, nodev) tmpfs -> /newroot/home/*/snap/steam/common/.local/share/vulkan/implicit_layer.d/,
+mount fstype=tmpfs options=(rw, nosuid, nodev) tmpfs -> /newroot/run/pressure-vessel/ldso/,
+mount fstype=tmpfs options=(rw, nosuid, nodev) tmpfs -> /newroot/tmp/.X11-unix/,
+
+# Pivot from the intermediate root to sandbox root
+mount options in (rw, silent, rprivate) -> /oldroot/,
+umount /oldroot/,
+pivot_root oldroot=/newroot/ /newroot/,
+umount /,
+
+# Permissions needed within sandbox root
+/usr/lib/pressure-vessel/** ixr,
+/run/host/** mr,
+/run/pressure-vessel/** mrw,
+/run/host/usr/sbin/ldconfig* ixr,
+/run/host/usr/bin/localedef ixr,
+/var/cache/ldconfig/** rw,
```

Comment:
Could we followup with that outside of this PR?  I'd like @jhenstridge to respond, but we won't get that as timely as we'd like.

Reply:
ah, these are in the pivoted root?  but is still mount /var from the snap root, so still not entirely sure how that is writable

Reply:
I added these rules to handle accesses made by the process within the mount namespace created by pressure-vessel/bubblewrap.

Maybe in future this could be separated out into a sub-profile, but that's complicated by the fact that the executables we'd perform the transitions on are downloaded by Steam and may have varying paths.

Reply:
I'm still a bit confused how that dir gets writable because the underlying dir comes from the base that is read-only

Reply:
The root of the bubblewrap sandbox is a tmpfs, so any path not mounted over is potentially writeable. It isn't exposing the whole host system `/run` or `/var`, so these locations are writeable (at least when AppArmor allows it).